### PR TITLE
new action : kill_processes

### PR DESCRIPTION
**Pull request by xpdable**

### CODE REVIEW

File: chaossaltstack/machine/actions.py [hunk 1]
```diff
@@ -602,6 +603,94 @@ def network_latency(instance_ids: List[str] = None,
...
+            param["instance_id"] = k
+            script_content = __construct_script_content__(
+                KILL_PROCESSES, os_type, param)
+
+            # Do async cmd and get jid
+            logger.debug("network_latency of machine: {}".format(name))
+            salt_method = 'cmd.run'
+            jid = client.async_run_cmd(name, salt_method, script_content)
+            jids[k] = jid
+        logger.debug(json.dumps(jids))
```
Comment (Lawouach) [inline chaossaltstack/machine/actions.py]: It's good practice to add a message with a log, so one can understand the context a bit more :)

File: chaossaltstack/machine/actions.py [hunk 2]
```diff
@@ -602,6 +603,94 @@ def network_latency(instance_ids: List[str] = None,
...
+        machines = client.get_grains_get(instance_ids, 'kernel')
+
+        param = dict()
+        param["duration"] = execution_duration
+        param["param"] = process_name
+
+        jids = dict()
+
+        if len(machines) <= 0:
+            FailedActivity(
```
Comment (Lawouach) [inline chaossaltstack/machine/actions.py]: you forgot to `raise`

Comment (Lawouach) [reply chaossaltstack/machine/actions.py]: probably add a tests for this too :)

File: chaossaltstack/machine/scripts/kill_processes.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+if [ -z "$process_name" ]; then
+    echo "Please provide process name."
+    exit 1
+fi
+
+experiment=$(killall -9 $process_name)
```
Comment (Lawouach) [inline chaossaltstack/machine/scripts/kill_processes.sh]: would it make sense to support killing with a different flag? maybe send a SIGINT? could this be a parameter?

File: chaossaltstack/machine/actions.py (hunk 1 above)
Comment (xpdable) [reply chaossaltstack/machine/actions.py]: As I find there are too many duplicates in each experiment acton, I will abstract the workflow to one parameterized internal one.

File: chaossaltstack/machine/scripts/kill_processes.sh (hunk 3 above)
Comment (xpdable) [reply chaossaltstack/machine/scripts/kill_processes.sh]: ok, let me try and test one AWS as well

**Comment by xpdable** [comment]
> Hi!
> 
> Thanks a lot for the PR. I was wondering, you are using scripts because you send them remotely to be executed on a remote machine? Or is the toolkit local to the scripts's execution?
> 
> If the toolkit is local to process being killed for instance, I would prefer a more portable solution like psutil: https://psutil.readthedocs.io/en/latest/#find-process-by-name and https://psutil.readthedocs.io/en/latest/#kill-process-tree
> 
> Can you clarify how this PR is meant to work please?
> 
> Thanks:

Hey, the action is used to kill a process on the remote machines. We use this function already in AWS to kill a process, e.g. a java thread to stop a service.

**Comment by Lawouach** [comment]
Okay I see. It seems the way to perform operations on a remote machine is to do it via the saltstack API, right? That API takes a script to execute on the remote machine, correct?

**Comment by xpdable** [comment]
yes. The saltstack API that we called to SaltMaster will distribute the script to SaltMinion (== the remote agent/machine). The mechanism is as same as AWS System Manager [run command](https://docs.aws.amazon.com/systems-manager/latest/userguide/execute-remote-commands.html) and Azure vm [run command](https://docs.microsoft.com/en-us/rest/api/compute/virtual%20machines%20run%20commands/runcommand)

**Comment by xpdable** [comment]
@Lawouach rebuild and passed CI on python3.5 :-P

**Pull request merged**